pkg/api/types.go

```diff
@@ -75,6 +78,25 @@ type (
 	ResourceThresholds map[v1.ResourceName]Percentage
 )
 
+// Negative returns a new ResourceThresholds with all its values negated.
+func (r ResourceThresholds) Negative() ResourceThresholds {
```

Comment:
`Negative` and `Round` are two examples of methods iterating through the list of floats64 here. If we abstract from the percentages and promote it into a map of float64 number in general, the type can be extend with `Ceil` and `Floor` methods that could round any negative number to 0 and any number higher than 100 (or 1) to 100 (1).

Reply:
Also, in general Kubernetes types that are part of a data API are not extended with methods. Rather, functions that take the types as arguments is the way to go.

Reply:
I think I achieved what you meant. Please take a look.